pkg/x509.go

```diff
@@ -108,15 +108,31 @@ func (client *Crypto) StoreVendorCACertificate(certificate *x509.Certificate) er
 	}
 	identity := core.NutsConfig().VendorID()
 	log.Logger().Infof("Storing CA certificate for: %s", identity)
+
 	key := types.KeyForEntity(types.LegalEntity{URI: identity.String()})
 	if !client.Storage.PrivateKeyExists(key) {
 		return fmt.Errorf("private key not present for key: %s", key)
 	}
+
 	if publicKey, err := client.Storage.GetPublicKey(key); err != nil {
 		return err
 	} else if !reflect.DeepEqual(publicKey, certificate.PublicKey) {
 		return fmt.Errorf("public key in certificate does not match stored private key (key: %s)", key)
 	}
+
+	// check ExtKeyUsage
+	expectedEKU := []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
+	found := 0
+	for _, exp := range expectedEKU {
+		for _, eku := range certificate.ExtKeyUsage {
+			if exp == eku {
+				found++
+			}
+		}
+	}
+	if found != 2 {
+		return fmt.Errorf("certificate does not define ExtKeyUsage: ClientAuth, ServerAuth")
```

Comment:
`fmt.Errorf("certificate does not define ExtKeyUsage: %v", expectedEKU)`

Reply:
x509.ExtKeyUsage are just numbers, so the error will print `[2 1]`

Reply:
KeyUsage is a bitmask which Go renders pretty nice (as OR'd components of the bitmask), but numbers it is, I guess